src/navigation/AppNavigator.tsx

```diff
@@ -0,0 +1,36 @@
+import { createNativeStackNavigator } from '@react-navigation/native-stack';
+import React, { useEffect, useState } from 'react';
+import { APP_NAME, SPLASH_SCREEN_TIME } from '../constants/AppConstant';
+import Homepage from '../screens/homepage';
+import SplashScreen from '../screens/splash';
+
+const Stack = createNativeStackNavigator();
+
+const AppNavigator = () => {
+    const [loading, setLoading] = useState(true);
+
+    useEffect(() => {
+        setTimeout(() => {
+            setLoading(false);
+        }, SPLASH_SCREEN_TIME);
+    }, []);
```

Comment:
Check once if we can remove the useEffect and directly call the setTimeout here. 
```suggestion
        setTimeout(() => {
            setLoading(false);
        }, SPLASH_SCREEN_TIME);
```

Reply:
Fixed in the recent commit. Thank you.